Automattic/wp-calypso
# Site Editor Tracking: Track browsing sidebar actions

#### Changes proposed in this Pull Request

* Track "Back to dashboard button click" in browsing sidebar `wpcom_block_editor_close_click`
* Track "Browsing Sidebar Open" action `wpcom_block_editor_nav_sidebar_open`
* Track "Template creation via browsing sidebar" `wpcom_block_editor_nav_sidebar_item_add`
  * Includes two event properties:
    * `item_type`: This is always `template`, indicating a template is being created
    * `item_slug`: Slug of the template
* Track "Template creation via browsing sidebar" `wpcom_block_editor_nav_sidebar_item_edit`
  * Includes two event properties:
    * `item_type`: This is either `template` or `template_part`, indicating what type of entity being switched to
    * `item_id`: ID of the template / template part
    * `item_slug`: Slug of the template / template part

#### Testing instructions

* Follow steps to enable tracking debugging at PCYsg-nrf-p2
* Open Site Editor

1. **Open the browsing sidebar (W/your site logo button in the top left corner**
1.1 Confirm `wpcom_block_editor_nav_sidebar_open` is triggered

2. **In the Template submenu create a new "Search" template (or anything else that is available)**
2.1 Confirm `wpcom_block_editor_nav_sidebar_item_add` is triggered with `item_type = "template"` and `item_slug = "search"`


3. **Open the browsing sidebar again and click on Templates -> Index**
3.1 Confirm `wpcom_block_editor_nav_sidebar_item_edit` is triggered with `item_type = "template"`, `item_id = "YOUR_THEME//index"` and `item_slug = "index"`


4. **Open the browsing sidebar again and click on Template Parts -> Headers -> Header**
4.1 Confirm `wpcom_block_editor_nav_sidebar_item_edit` is triggered with `item_type = "template_part"` and `item_slug = "header"`


5. **Open the browsing sidebar and click on the "Back to dashboard" button**
5.1 Confirm `wpcom_block_editor_close_click` is triggered



Related to #53410
